spring-integration-amqp/src/main/java/org/springframework/integration/amqp/outbound/AbstractAmqpOutboundEndpoint.java

```diff
@@ -479,8 +480,10 @@ protected CorrelationData generateCorrelationData(Message<?> requestMessage) {
 			if (messageId == null) {
 				messageId = NO_ID;
 			}
-			correlationData = new CorrelationDataWrapper(messageId.toString(),
-					this.correlationDataGenerator.processMessage(requestMessage), requestMessage);
+			Object userData = this.correlationDataGenerator.processMessage(requestMessage);
+			if (userData != null) {
+				correlationData = new CorrelationDataWrapper(messageId.toString(), userData, requestMessage);
```

Comment:
Looks like previously we have had a `CorrelationDataWrapper` instance independently of the `userData`, but now you restrict it only if that one is not null.

Also I see that you are protected against `null` in the `CorrelationDataWrapper` any way.

So, what is the motivation to not do that any more?

Reply:
Because we get an NPE in `handleConfirm()` if the expression resolves to `null` (bug - try to create a message with a `null` payload).

I should have mentioned it in the commit comment and, I suppose, this part should be back-ported.

Reply:
Should we emit WARN  then in case of `correlationDataGenerator` resolves to `null`?

Reply:
I decided to go with DEBUG (there is a corresponding DEBUG in `handleConfirm`).

Someone might want to use this as a feature ("I don't care about a confirm for this message").